File: src/main/java/ee/taltech/gtm/CustomDocumentListener.java
package ee.taltech.gtm;

import com.intellij.openapi.editor.Document;
import com.intellij.openapi.editor.event.DocumentEvent;
import com.intellij.openapi.editor.event.DocumentListener;
import com.intellij.openapi.fileEditor.FileDocumentManager;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;

public class CustomDocumentListener implements DocumentListener {

    @Override
    public void documentChanged(@NotNull DocumentEvent event) {
        Document document = event.getDocument();
        FileDocumentManager instance = FileDocumentManager.getInstance();
        VirtualFile file = instance.getFile(document);
        var project = Util.getProject(document);
        project.ifPresent(p -> GtmWrapper.getInstance().recordFile(p, file));
    }
}

File: src/main/java/ee/developest/gtm/listener/CustomDocumentListener.kt
package ee.developest.gtm.listener

import com.intellij.openapi.editor.event.DocumentEvent
import com.intellij.openapi.editor.event.DocumentListener
import com.intellij.openapi.fileEditor.FileDocumentManager
import com.intellij.openapi.project.Project
import ee.developest.gtm.GtmWrapper
import ee.developest.gtm.Util

class CustomDocumentListener : DocumentListener {
    override fun documentChanged(event: DocumentEvent) {
        val document = event.document
        val instance = FileDocumentManager.getInstance()
        val file = instance.getFile(document) ?: return
        val project = Util.getProject(document)
        project.ifPresent { p: Project? -> GtmWrapper.instance.recordFile(p, file) }
    }


}